lib/rucio/core/lock.py

```diff
@@ -137,15 +137,25 @@ def get_replica_locks_for_rule_id(rule_id, session=None):
 
     locks = []
 
-    query = session.query(models.ReplicaLock).filter_by(rule_id=rule_id)
-
-    for row in query:
-        locks.append({'scope': row.scope,
-                      'name': row.name,
-                      'rse_id': row.rse_id,
-                      'rse': get_rse_name(rse_id=row.rse_id, session=session),
-                      'state': row.state,
-                      'rule_id': row.rule_id})
+    query = session.query(models.ReplicaLock, models.Request).join(
```

Comment:
Hi JJ, I don't think it will work like this.
- The `get_replica_locks_for_rule_id` call should probably also return locks, if you don't have any requests, so you need to do a left outer join, otherwise it won't work.
- The join condition on rule_id, while in most cases this would work, in some it will not, because for some locks the request is actually from a different rule, so it won't match. you have to do the `ON` filter of the join on `replicaLock.scope=request.scope and replicalock.name=request.name`. But the filter condition can be on replicalock.rule_id, that is ok. 
Otherwise it is good, although a test wood be nice.